crates/jstz_api/src/file/blob.rs

```diff
@@ -0,0 +1,511 @@
+//! `jstz`'s implementation of JavaScript's `Blob` API.
+//!
+//! More information:
+//!  - [MDN documentation][mdn]
+//!  - [W3C `File` specification][spec]
+//!  - [WHATWG `Infra` specification][infra-spec]
+//!
+//! [mdn]: https://developer.mozilla.org/en-US/docs/Web/API/Blob
+//! [spec]: https://w3c.github.io/FileAPI/
+//! [infra-spec]: https://infra.spec.whatwg.org/
+
+use crate::idl::{ArrayBufferLike, JsBufferSource};
+use boa_engine::{
+    js_string,
+    object::{
+        builtins::{JsArray, JsArrayBuffer, JsPromise},
+        Object,
+    },
+    property::Attribute,
+    value::TryFromJs,
+    Context, JsArgs, JsError, JsNativeError, JsResult, JsString, JsValue, NativeFunction,
+};
+use boa_gc::{Finalize, GcRefMut, Trace};
+use jstz_core::{
+    accessor,
+    native::{register_global_class, Accessor, JsNativeObject, NativeClass},
+    value::IntoJs,
+};
+use std::cmp::{max, min};
+
+#[derive(Trace, Finalize, Clone)]
+pub struct Blob {
+    bytes: Vec<u8>,
+    size: u64,
+    type_: String,
```

Comment:
Internally I'd go for `Option<String>` since the default value of this is `""`. Using `None` over `""` makes it clearer (at least to me)

Reply:
Not sure how to reconcile using `Option<String>` with the places in the API spec. in which it talks about the "empty string". Should I use `None` or `Some(String::new())`?

Reply:
`None` for empty string